components/dialog/dialog.js

```diff
@@ -158,7 +158,6 @@ class Dialog extends LocalizeCoreElement(AsyncContainerMixin(DialogMixin(LitElem
 			'd2l-footer-no-content': !this._hasFooterContent
 		};
 
-		if (!this._textId && this.describeContent) this._textId = getUniqueId();
```

Comment:
Again, I think we should just always set `this._textId` in the constructor to avoid this situation. In this case, we'd be setting it sometimes when we don't need it (i.e. if `this.describeContent` is `false`), but it's cheap to generate an ID even if it doesn't get used.